packages/core/src/factory.ts

```diff
@@ -14,60 +18,66 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
   defaultConfig: Config,
   components: IPrismComponents<Resource, Input, Output, Config>
 ): IPrism<Resource, Input, Output, Config> {
+  type ResourceAndValidation = {
+    resource: Resource;
+    inputValidations: IPrismDiagnostic[];
+  };
+
+  const inputValidation = (
```

Comment:
something more verb-ish, like:

```suggestion
  const getInputValidation = (
```